docs/changelog/93396.yaml

```diff
@@ -0,0 +1,51 @@
+pr: 93396
+summary: Add support for Reciprocal Rank Fusion to the search API
+area: Ranking
+type: feature
+issues: []
+highlight:
+  title: Add support for Reciprocal Rank Fusion to the search API
+  body: |-
+    This change at a high level adds global re-ranking on the coordinating
+    node at the end of query reduction prior to the fetch phase. The
+    singular re-ranking model added in this change is reciprocal rank fusion
+    (RRF) which follows the basic formula for merging `1...n` sets of
+    results sets together with `sum(1/(k+d))` where k is a ranking constant
+    and d is a document's scored position within a result set from a query.
+
+    The API for this change adds a `rerank` top-level element to the search
+    endpoint. An example:
+
+    [source,Java]
+    ----
+    {
+      "query": {
+        "match": {
+          "product": {
+            "query": "brown shoes"
+          }
+        }
+      },
+      "knn": {
+        "field": "product-vector",
+        "query_vector": [54, 10, -2],
+        "k": 20,
+        "num_candidates": 75
+      },
+      "rank": {
+         "rrf": {
+            "window_size": 100,
+            "rank_constant": 20
+         }
+      }
+    }
+    ----
+
+    The above example will execute two separate queries for the match query
+    and the kNN query. It will preserve separate result sets up the point
```

Comment:
From this explanation, as an advanced user I may wonder where the separate queries are executed, how this affects the search roundtrips. Instead of expanding this though, I wonder if we should keep the highlight description more high-level and rather explain what RRF helps with: poorly said,  it mixes together docs that have scores on different scales?

Reply:
Docs need work for sure. I've still only stubbed out the RRF docs.